pkgs/by-name/st/startup-disk/package.nix

```
@@ -0,0 +1,67 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitLab,
+  nix-update-script,
+  pkg-config,
+  libadwaita,
+  gtk4,
+  glib,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "startup-disk";
+  version = "0.1.5";
+
+  src = fetchFromGitLab {
+    owner = "davide125";
+    repo = "startup-disk";
+    tag = "${version}";
+    hash = "sha256-258whEX6hKqfrk2aII15tuFEuB7NQUCNLEmi3OCOWV4=";
+    domain = "gitlab.gnome.org";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    glib.dev
```

```suggestion
    glib
```

`.dev` shouldn't be necessary

---

> GLib is pulled in transitively by libadwaita

It doesn't work like that, all dependencies need to be explicitly listed

> is it better to specifiy it explicitly in `buildInputs`?

Runtime dependencies → goes to `buildInputs`

One way to think about this is in terms of cross compilation. If you're cross compiling from aarch64 to x86_64, then, very roughly speaking, `nativeBuildInputs` is the aarch64 stuff, and `buildInputs` is the x86_64 stuff.

---

Ah okay, thank you for the explanation! 

The program fails to build when glib is only specified in buildInputs, does it make sense to specify in both buildInputs and nativeBuildInputs?